### PR TITLE
Add unsecure proxy

### PR DESCRIPTION
**Pull request by mikaelpopowicz**
Updates proxy command to handle a `--unsecure` option. As you may want unsecured Valet site you may want unsecured Valet proxy.

`valet proxy [domain] [host] [--unsecure]`

eg:

`valet proxy docker http://127.0.0.1:8080 --unsecure` creates a config file for `docker.test`, which proxies all traffic for `http://docker.test` through to `http://127.0.0.1:8080`

--

Note: To return to a secured proxy, `proxy` again the same domain/host tuple without `--unsecure` option.

### CODE REVIEW

**Comment by drbyte** [comment]
If the consensus is to go ahead with this, then I suggest that the extra "port 60" `server` sections be removed from both proxy stubs. Those are intended for ngrok when serving direct sites. (Yes, probably should have been removed earlier; but if these updated stubs are the new way going forward then this is a good time to clear the extra section out.)

**Comment by mikaelpopowicz** [comment]
Hello @drbyte

Should I update the PR accordingly to #1013 ?

**Comment by drbyte** [comment]
> Should I update the PR accordingly to #1013 ?

It's a valid point in terms of consistency. But adds a lot more code complexity.

**Comment by Norgul** [comment]
This is just what I needed! Thanks

**Comment by mattstauffer** [comment]
Sorry for the delay on this; @mikaelpopowicz would you mind fixing it to avoid the conflicts, and then i'll merge and release?

Thanks!

**Comment by mikaelpopowicz** [comment]
Yes

**Comment by mikaelpopowicz** [comment]
I just realize I put back Ngrok stuff. I'll remove it again soon.

File: cli/Valet/Site.php
Comment (mattstauffer) [inline cli/Valet/Site.php:729]: I wonder, would it make sense to name this `insecure.proxy.valet.conf` to make it super clear?

File: cli/Valet/Site.php
Comment (mattstauffer) [inline cli/Valet/Site.php:780]: So, because this has `unsecure` in it, I think we may need to rename it. It may have to be something like `putInsecurely` or something? I know that's awkward, but the idea that this thing that's simply "putting" it *also* insecures it seems a bit unexpected. That said, I'm not *super* familiar with this class; @drbyte do you have other suggestions?

**Review by mattstauffer — CHANGES_REQUESTED** [review]
Everything else looks great other than these questions, which I think I'll need @drbyte to weigh in on. Thanks so much for your work on this @mikaelpopowicz!

File: cli/Valet/Site.php
Comment (mikaelpopowicz) [reply cli/Valet/Site.php:716]: Ok, I'll update the parameter and variable.

Just a question, `insecure` may seem pejorative. Maybe it should be about protocol : http v https ?

Comment (drbyte) [reply cli/Valet/Site.php]: If I'm honest, I don't like this change to `insecure` at all. I'd rather we rollback this latest.

Valet's convention for other sites (not proxy), is `valet.conf` and `secure.valet.conf`. When it handles securing something, that's when it uses wording/terms related to `secure`. Otherwise it assumes not-secure.

I did break that convention slightly when I added the proxy option, *specifically* because of all the complexities that this PR highlights. But, here we are. 
If we're going ahead with this PR, then I'd suggest rolling back all these `insecure` rewording changes.

Comment (mattstauffer) [reply cli/Valet/Site.php]: @drbyte can you explain that a bit?

I think there are *two* changes I suggested here:

1. prefixing `valet.conf` with `insecure`, which, if it breaks convention, is totally fine to roll back!
2. Changing `unsecure` to `insecure` *if* this is referring to the word being used as a *description* (adjective) instead of a *command* (verb).

Sounds like we should rollback the prefixing of the file.

Can anyone explain to me: is `unsecure` an action we're taking? "Unsecure this site!" ? Or is it a description of *how* we're doing something, adverbially? "Do this action, but in an insecure way". That is what defines which word it should be for me.

Comment (drbyte) [reply cli/Valet/Site.php]: In a general sense, if we think about the concept "outside of this proxy-related PR", Valet uses the words "secure" and "unsecure" as actions, to add/remove SSL on a site.

Internally some class methods therefore are related to actions taken to fulfill those requests.
In practice, the action of "creating a proxy-site using a non-secured proxy stub" must for safety/cleanup reasons first ensure that the site is `unsecure`'d in case that site had been "secured" in the past. So `unsecure` as a verb for the method call makes sense.
Thus I don't think we need `putInsecurely()` etc. IMO `put()` is fine, and `secure()`/`unsecure()` are fine.

Where `$unsecure` is used as a method *parameter* (such as in `proxyCreate()`, maybe we can rename it to `$secure` and reverse the logic accordingly, to minimize confusion.

As for stub filename, I don't think we need the `insecure.` prefix on the filename.

Comment (mattstauffer) [reply cli/Valet/Site.php]: - I'm totally fine without the prefix on the filename
- I definitely think `secure` and `unsecure` are good method names
- totally get the reason for wanting to call `unsecure` before creating a proxy-site using a non-secured proxy stub, and therefore totally understand the not needing to rename to `putInsecurely`
- I'm OK with renaming it to `$secure` in the method list and then inverting it, think that's a great idea

That only leaves the actual command line flag, which I still think makes sense to be `--insecure`, but what do you think?

Comment (drbyte) [reply cli/Valet/Site.php]: How about `--secure` for CLI too? ie: defaults to non-secure, unless --secure is specified?

Comment (mattstauffer) [reply cli/Valet/Site.php]: hm, that's a good point! It's better language *and* behaves more than the rest of the site.

I think the only downside would be, if I have my app secured, and then I run `valet share`, I don't think I would expect it to suddenly un-secure it.. right? that's a problem either way though. I AM LOST AND CONFUSED

Comment (drbyte) [reply cli/Valet/Site.php]: I thought we were only talking about `valet proxy --secure`, not `valet share`

Comment (mattstauffer) [reply cli/Valet/Site.php]: @drbyte Oh crap. You're absolutely right. YES. I like that. Sorry!

**Comment by drbyte** [comment]
Thanks @mikaelpopowicz !

**Comment by mikaelpopowicz** [comment]
You're welcome @drbyte 
Hi @mattstauffer the workflow is missing approval to run.

File: cli/stubs/secure.valet.conf
Comment (drbyte) [inline cli/stubs/secure.valet.conf:53]: @mikaelpopowicz I think this port 60 part needs to stay in this file, which essentially then also removes this file from the PR.
(The port 60 stuff is used by ngrok when running the valet share command.)

Comment (mikaelpopowicz) [reply cli/stubs/secure.valet.conf:53]: Yes, it was a mistake.

**Comment by drbyte** [comment]
@mattstauffer if you can kick off the tests (the GH pending workflow thing), 
then I think this is ready for merge.

**Review by drbyte — APPROVED** [review]

**Comment by mattstauffer** [comment]
Thanks @mikaelpopowicz and @drbyte! Running the tests and will merge and tag if they're green. Thanks!

**Pull request merged**
